src/OrleansRuntime/Core/InsideRuntimeClient.cs

```diff
@@ -414,6 +428,36 @@ public void SniffIncomingMessage(Message message)
             }
         }
 
+        private static void UnwrapCancellationToken(IAddressable target, object arg, InvokeMethodRequest request, int i)
+        {
+            var grainToken = ((GrainCancellationToken)arg);
+            if (grainToken.WentThroughSerialization)
+            {
+                CancellationSourcesExtension cancellationExtension;
+                if (!SiloProviderRuntime.Instance.TryGetExtensionHandler(out cancellationExtension))
+                {
+                    cancellationExtension = new CancellationSourcesExtension();
+
+                    if (!SiloProviderRuntime.Instance.TryAddExtension(cancellationExtension))
+                    {
+                        if (logger.IsWarning) logger.Warn(
+                            ErrorCode.CancellationExtensionCreationFailed,
+                            string.Format("Could not add cancellation token extension, target: {0}", target));
+                        return;
```

Comment:
Acually it shouldn't be called even once, just didn't wanted to ignore theoretical failure of `TryAddExtension` operation.